docs/data/data-grid/server-side-data/ServerSideLazyLoadingErrorHandling.tsx

```diff
@@ -0,0 +1,111 @@
+import * as React from 'react';
+import {
+  DataGridPro,
+  useGridApiRef,
+  GridToolbar,
+  GridDataSource,
+  GridGetRowsParams,
+  GRID_ROOT_GROUP_ID,
+} from '@mui/x-data-grid-pro';
+import Checkbox from '@mui/material/Checkbox';
+import FormControlLabel from '@mui/material/FormControlLabel';
+import { useMockServer } from '@mui/x-data-grid-generator';
+import Alert from '@mui/material/Alert';
+import Button from '@mui/material/Button';
+
+function ErrorAlert({ onClick }: { onClick: () => void }) {
+  return (
+    <Alert
+      sx={{
+        position: 'absolute',
+        bottom: '0',
+        paddingX: 2,
+        paddingY: 1,
+        width: '100%',
+        zIndex: 10,
+      }}
+      severity="error"
+      action={
+        <Button color="inherit" size="small" onClick={onClick}>
+          Retry
+        </Button>
+      }
+    >
+      Could not fetch the data
+    </Alert>
+  );
+}
+
+function ServerSideLazyLoadingErrorHandling() {
+  const apiRef = useGridApiRef();
+  const [retryParams, setRetryParams] = React.useState<GridGetRowsParams | null>(
```

Comment:
Is it possible to hide the Alert when the data is fetched? Maybe by using `rowsFetched` event?

https://github.com/user-attachments/assets/30f280ea-d9eb-4b3d-a8f9-27e1cd726465

That being said, I think passing `getRowsParams` as additional information with the `rowsFetched` event could be useful.

What do you think?

Reply:
Done.
Actually, events are not needed and I have removed them completely.
Instead of triggering new fetch trough event, it is done via data source API.
And rows fetched is also not needed, because you can handle that in your `getRows` callback (like in the example now)